libs/cart/driver/src/errors/invalid-coupon-code.ts

```diff
@@ -12,7 +13,7 @@ import { DaffCartDriverErrorCodes } from './codes.enum';
 export class DaffInvalidCouponCodeError extends DaffInheritableError implements DaffError {
 	public readonly code: string = DaffCartDriverErrorCodes.INVALID_COUPON_CODE;
 
-	constructor(message?: string) {
+	constructor(message?: string, public coupon?: DaffCartCoupon['code']) {
```

Comment:
I think we should actually modify the error message instead. If we add this extra field, we'll end up needing to do the same thing in every platform coupon driver, so the app dev won't need to deal with the differences between the `DaffInvalidCouponCodeError`s when they change platforms. So maybe just append `coupon.code + " coupon code is invalid"` to the actual message.

Reply:
> So maybe just append coupon.code + " coupon code is invalid" to the actual message.

That makes it significantly harder to use the coupon code in a custom error message, which is the use case being covered by this PR. I think it should stay as a field on the error.

> we'll end up needing to do the same thing in every platform coupon driver

This is preferable to forcing the app dev to regex the coupon code out of the error message.

Reply:
@lderrickable why would there be a difference in DaffInvalidCouponError between platforms?

Reply:
@damienwebdev Because if Shopify returns the coupon code in the error message itself, then the Daffodil error wouldn't need the `public coupon?: DaffCartCoupon['code']` to pass on the coupon code information. So the app dev would need to stop using the `coupon` field, because that would now be `null`.

Reply:
The use case in question is that the app dev wants to customize the error message in a way that requires having the coupon code. Regardless of whether or not the code is in the message, I think it should be set as a field on the error. The app dev should not be forced to regex the code out of the error message.

Reply:
@griest024 oh ok. If we want to allow the app dev to use the coupon code in a custom error message, then I actually think it makes more sense for this to go in @daffodil/state, since we'll need the coupon field to be separate for every platform. Sorry for confusion. I guess I didn't understand the purpose of this PR.

Reply:
Ok, I thought about this again and convinced myself that it makes more sense for this to be in the driver